src/components/templates/TemplateWorkflowsContent.vue

```diff
@@ -1,68 +1,127 @@
+                  >
+                    <i class="pi pi-play-circle" style="color: white"></i>
+                  </div>
+                </a>
+                <ProgressSpinner
+                  v-if="loading === slotProps.data"
+                  class="absolute inset-0 z-1 w-3/12 h-full"
+                />
               </div>
-            </a>
-            <ProgressSpinner
-              v-if="loading === template"
-              class="absolute inset-0 z-1 w-3/12 h-full"
-            />
-          </div>
-        </template>
-        <template #subtitle>{{
-          $t(`templateWorkflows.template.${template}`)
-        }}</template>
-      </Card>
-    </div>
+            </div>
+          </template>
+          <template #subtitle>{{ slotProps.data }}</template>
+        </Card>
+      </template>
+    </Carousel>
   </div>
 </template>
 
 <script setup lang="ts">
 import { useDialogStore } from '@/stores/dialogStore'
+import Carousel from 'primevue/carousel'
+import Listbox from 'primevue/listbox'
 import Card from 'primevue/card'
 import ProgressSpinner from 'primevue/progressspinner'
-import { ref } from 'vue'
+import { onMounted, ref } from 'vue'
 import { app } from '@/scripts/app'
 import { api } from '@/scripts/api'
 import { useI18n } from 'vue-i18n'
+import toast from 'primevue/toast'
 const { t } = useI18n()
 
-const templates = ['default', 'image2image', 'upscale', 'flux_schnell']
 const loading = ref<string | null>(null)
 
+interface WorkflowTemplatesTab {
+  moduleName: string
+  title: string
+  templates: string[]
+}
+
+//These default templates are provided by the frontend
+const comfyUITemplates = {
+  moduleName: 'default',
+  title: 'ComfyUI',
+  templates: ['default', 'image2image', 'upscale', 'flux_schnell']
+}
+
+const tabs = ref<WorkflowTemplatesTab[]>([comfyUITemplates])
+
+const selectedTab = ref<WorkflowTemplatesTab>(comfyUITemplates)
+
+onMounted(async () => {
+  try {
+    const workflowTemplates = await api.getWorkflowTemplates()
+    tabs.value = [
+      comfyUITemplates,
+      ...Object.entries(workflowTemplates).map(([key, value]) => ({
+        moduleName: key,
+        title: key,
+        templates: value
+      }))
+    ]
+  } catch (error) {
+    console.error('Error fetching workflow templates:', error)
+    toast.add({
+      severity: 'error',
+      summary: 'Error',
+      detail: 'Failed to fetch workflow templates',
+      life: 5000
+    })
+  }
+})
+
 const loadWorkflow = async (id: string) => {
   loading.value = id
-  const json = await fetch(api.fileURL(`templates/${id}.json`)).then((r) =>
-    r.json()
-  )
+  let json
+  if (selectedTab.value.moduleName === 'default') {
+    // Default templates provided by frontend are served on this separate endpoint
+    json = await fetch(api.fileURL(`templates/${id}.json`)).then((r) =>
+      r.json()
+    )
+  } else {
+    json = await fetch(
+      api.fileURL(
+        `workflow_templates/${selectedTab.value.moduleName}/${id}.json`
```

Comment:
In the associated backend PR:

```python
webapp.add_routes([web.static('/api/workflow_templates/' + module_name, workflows_dir)])
```

It may be better to drop the `api` prefix since these are static files. Otherwise, I believe the prefix must be added manually here in the `api.fileURL` call 👍.

Reply:
I read in a code comment that the routes without the `api/` are old but left around for a while before they're removed.  that's why I put everything under `api/` and I've changed the above code to call `apiURL`, should work now